kotlinpoet/src/jvmMain/kotlin/com/squareup/kotlinpoet/AnnotationSpec.kt

```diff
@@ -202,7 +202,7 @@ public class AnnotationSpec private constructor(
         @Suppress("PLATFORM_CLASS_MAPPED_TO_KOTLIN")
         val javaAnnotation = annotation as java.lang.annotation.Annotation
         val builder = builder(javaAnnotation.annotationType())
-          .tag(annotation)
+          .tag<Annotation>(annotation)
```

Comment:
Fixes the following warning:

```
e: warnings found and -Werror specified
w: file:///Users/egor/Development/block-workspace/kotlinpoet/kotlinpoet/src/jvmMain/kotlin/com/squareup/kotlinpoet/AnnotationSpec.kt:205:12 
Type argument for reified type parameter 'T' was inferred to the intersection 
of ['java.lang.annotation.Annotation' & 'kotlin.Annotation']. 
Reification of an intersection type results in the common supertype being used. 
This may lead to subtle issues and an explicit type argument is encouraged. 
This will become an error in a future release.
```